packages/nameguard-react/src/components/Report/RatingIcon.tsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import { Rating } from "@namehash/nameguard";
+
+import { RatingPassLargeIcon } from "../icons/RatingPassLargeIcon";
+import { RatingWarnLargeIcon } from "../icons/RatingWarnLargeIcon";
+import { RatingAlertLargeIcon } from "../icons/RatingAlertLargeIcon";
+import { RatingPassMediumIcon } from "../icons/RatingPassMediumIcon";
+import { RatingWarnMediumIcon } from "../icons/RatingWarnMediumIcon";
+import { RatingAlertMediumIcon } from "../icons/RatingAlertMediumIcon";
+import { RatingPassSmallIcon } from "../icons/RatingPassSmallIcon";
+import { RatingWarnSmallIcon } from "../icons/RatingWarnSmallIcon";
+import { RatingAlertSmallIcon } from "../icons/RatingAlertSmallIcon";
+import { RatingAlertMicroIcon } from "../icons/RatingAlertMicroIcon";
+import { RatingWarnMicroIcon } from "../icons/RatingWarnMicroIcon";
+import { RatingPassMicroIcon } from "../icons/RatingPassMicroIcon";
+
+export enum RatingIconSize {
+  micro = "micro",
+  small = "small",
+  medium = "medium",
+  large = "large",
+}
+
+type Props = {
+  rating: Rating;
+  size?: RatingIconSize;
+  className?: string;
+};
+
+const STATUS_TO_BASE_NAME: { [key in Rating]: string } = {
+  [Rating.alert]: "RatingAlert",
+  [Rating.pass]: "RatingPass",
+  [Rating.warn]: "RatingWarn",
+};
+
+const getComponent = (rating: Rating, size: RatingIconSize): React.Node => {
+  const baseName = STATUS_TO_BASE_NAME[rating];
+  const componentName = `${baseName}${capitalizeFirstLetter(size)}Icon`;
+
+  const components: { [key: string]: React.ComponentType } = {
+    RatingPassLargeIcon,
+    RatingPassMediumIcon,
+    RatingPassSmallIcon,
+    RatingPassMicroIcon,
+    RatingWarnLargeIcon,
+    RatingWarnMediumIcon,
+    RatingWarnSmallIcon,
+    RatingWarnMicroIcon,
+    RatingAlertLargeIcon,
+    RatingAlertMediumIcon,
+    RatingAlertSmallIcon,
+    RatingAlertMicroIcon,
+  };
+
+  if (!components[componentName]) {
+    throw new Error(
+      `RatingIcon could not be built with params: \n\n rating: ${rating}\n size: ${size}`,
+    );
+  }
+
+  return components[componentName];
+};
+
+function capitalizeFirstLetter(string: string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+export const RatingIcon = ({
```

Comment:
Please note how `RatingIcon` is supposed to have a hover state for all cases except for loading.

https://www.figma.com/file/Kt4nUHyZJ5NuPCSZZkXmPL/NameGuard?type=design&node-id=395-3753&mode=design&t=wrzBFLo8iSoUN5bc-4

![CleanShot 2024-05-07 at 17 46 24](https://github.com/namehash/nameguard/assets/126201998/92b54453-a98d-4dc7-a192-b7c62781c426)

Could we please add these? Thank you :+1:

Reply:
There is one misunderstanding above: the artboard mentioned describes `ReportIcon` and not `RatingIcon`

`RatingIcon` represents three different states: Rating.pass, Rating.warn and Rating.alert
`ReportIcon` renders `RatingIcon` and, in case of a failed rating status or a loading status, displays the Unknown state or the Loading state.

Reply:
Hover states added 😉